File: app/src/main/java/ru/aradxxx/ciceronetabssample/di/FragmentBindingModule.java
package ru.aradxxx.ciceronetabssample.di;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import ru.aradxxx.ciceronetabssample.tabfragment.Tab1;
import ru.aradxxx.ciceronetabssample.tabfragment.Tab2;
import ru.aradxxx.ciceronetabssample.tabfragment.Tab3;
import ru.aradxxx.ciceronetabssample.tabfragment.TabContainerFragment;

@Module
public interface FragmentBindingModule {
    @ContributesAndroidInjector
    Tab1 provideTab1Fragment();

    @ContributesAndroidInjector
    Tab2 provideTab2Fragment();

    @ContributesAndroidInjector
    Tab3 provideTab3Fragment();

    @ContributesAndroidInjector
    TabContainerFragment provideTabContainerFragment();
}

File: app/src/main/java/com/github/aradxxx/ciceroneflowssample/di/FragmentBindingModule.kt
package com.github.aradxxx.ciceroneflowssample.di

import dagger.Module
import dagger.android.ContributesAndroidInjector
import com.github.aradxxx.ciceroneflowssample.tabfragment.Tab1
import com.github.aradxxx.ciceroneflowssample.tabfragment.Tab2
import com.github.aradxxx.ciceroneflowssample.tabfragment.Tab3
import com.github.aradxxx.ciceroneflowssample.tabfragment.TabContainerFragment

@Module
interface FragmentBindingModule {
    @ContributesAndroidInjector
    fun provideTab1Fragment(): Tab1

    @ContributesAndroidInjector
    fun provideTab2Fragment(): Tab2

    @ContributesAndroidInjector
    fun provideTab3Fragment(): Tab3

    @ContributesAndroidInjector
    fun provideTabContainerFragment(): TabContainerFragment
}
